doc/design/cluster_train/trainer.md

```diff
@@ -0,0 +1,24 @@
+## Design Doc : Trainer 
+
+the trainer role in whole system please refer to [distributed training design doc](./README.md). 
+
+This design doc only focus on Master Server and Trainer synchronization and python client event processing . The Task dispatch interface please refer to the [master_server](./master_server.md), [data_dispatch](./data_dispatch.md) and so on.
+
+## Synchronize SGD
+
+In synchronize SGD, trainer need to wait other nodes finish training in every minibatch. And don't go on next epoch training if there is any node lag behind.
```

Comment:
Here are some term we usually use:
step: one forward backward step, computes gradient.
mini-batch: several data instances used in a single step.
task: multiple mini-batches, the master server assigns task to trainers.
pass: all training data, many tasks.
epoch: start of a new pass.

In this line, "epoch" is used with "mini-batch", I think you meant "step"?